lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -94,42 +94,74 @@ public function get_items( $request ) {
 			return new WP_Error( 'rest_no_search_term_defined', __( 'You need to define a search term to order by relevance.' ), array( 'status' => 400 ) );
 		}
 
-		$args                         = array();
-		$args['author__in']           = $request['author'];
-		$args['author__not_in']       = $request['author_exclude'];
-		$args['menu_order']           = $request['menu_order'];
-		$args['offset']               = $request['offset'];
-		$args['order']                = $request['order'];
-		$args['orderby']              = $request['orderby'];
-		$args['paged']                = $request['page'];
-		$args['post__in']             = $request['include'];
-		$args['post__not_in']         = $request['exclude'];
-		$args['name']                 = $request['slug'];
-		$args['post_parent__in']      = $request['parent'];
-		$args['post_parent__not_in']  = $request['parent_exclude'];
-		$args['post_status']          = $request['status'];
-		$args['s']                    = $request['search'];
+		$registered = $this->get_collection_params();
```

Comment:
@rmccue Seems like this check against the registered params should happen in `get_allowed_query_vars()` so it isn't overridden